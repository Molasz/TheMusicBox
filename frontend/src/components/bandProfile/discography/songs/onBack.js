function onBack(event) {
  event.preventDefault();
  dispatch(showDisc());
}

export default onBack;
