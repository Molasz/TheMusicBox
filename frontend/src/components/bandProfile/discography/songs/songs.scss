.songs {
  .songs__top {
    display: flex;

    .top__img {
      width: 200px;
      margin-right: 20px;
    }
  }
  .songs__middle {
    //list syles
  }
  .songs__bottom {
    display: flex;
    .item {
      width: 150px;
      margin-right: 15px;
    }
  }
}
