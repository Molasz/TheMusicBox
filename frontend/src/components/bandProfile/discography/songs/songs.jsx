import React from 'react';
import { connect } from 'react-redux';
import { showDisc } from '../../../../redux/actions/bandActions';

import './songs.scss';

import Disc from '../disc/disc';

import BackIcon from '@material-ui/icons/KeyboardBackspace';

function Songs({ info, index, dispatch }) {
  function onBack(event) {
    event.preventDefault();
    dispatch(showDisc());
  }
  return (
    <section className='songs'>
      <div className='songs__top'>
        <img src={info[index].img} alt='Album cover' className='top__img' />
        <div className='top__content'>
          <BackIcon
            className='content__icon'
            onClick={(event) => onBack(event)}
          />
          <div className='content__text'>
            <h3 className='text__title'>{info[index].title}</h3>
            <p className='text__info'>
              {info[index].date} | {info[index].songs.length} songs
            </p>
          </div>
        </div>
      </div>
      <div className='songs__middle'>
        {info[index].songs.map((element, i) => {
          return (
            <p className='middle__item' key={i}>
              <span className='item__number'>{i + 1}.</span>
              {element.title} | {element.time}
            </p>
          );
        })}
      </div>
      <div className='songs__bottom'>
        {info.map((element, i) => {
          if (i !== index) {
            return <Disc data={element} index={i} />;
          }
        })}
      </div>
    </section>
  );
}

export default connect(null, null)(Songs);
