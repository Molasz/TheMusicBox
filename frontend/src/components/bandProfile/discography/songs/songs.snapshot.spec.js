import renderer from 'react-test-renderer';
import React from 'react';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

import { Songs } from './songs';

import onBack from './onBack';
jest.mock('./onBack');

describe('Search snapshot', () => {
  const info = [
    {
      title: 'title',
      songs: [{ time: 'time', title: 'title' }],
      date: 'date',
      img: 'img'
    }
  ];
  const index = 0;
  it('Should match without band data', () => {
    const tree = renderer.create(<Songs info={info} index={index} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should call onBack when click', () => {
    const document = shallow(<Songs info={info} index={index} />);
    const button = document.find('.content__icon');
    button.simulate('click', { preventDefault: () => {} });

    expect(onBack.call).truthy;
  });
});
