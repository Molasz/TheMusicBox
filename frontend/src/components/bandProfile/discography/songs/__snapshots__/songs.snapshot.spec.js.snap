// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot Should match without band data 1`] = `
<section
  className="songs"
>
  <div
    className="songs__top"
  >
    <img
      alt="Album cover"
      className="top__img"
      src="img"
    />
    <div
      className="top__content"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root content__icon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
        />
      </svg>
      <div
        className="content__text"
      >
        <h3
          className="text__title"
        >
          title
        </h3>
        <p
          className="text__info"
        >
          date
           | 
          1
           songs
        </p>
      </div>
    </div>
  </div>
  <div
    className="songs__middle"
  >
    <p
      className="middle__item"
    >
      <span
        className="item__number"
      >
        1
        .
      </span>
      title
       | 
      time
    </p>
  </div>
  <div
    className="songs__bottom"
  >
    
  </div>
</section>
`;
